packages/dd-trace/src/telemetry/index.js

```diff
@@ -322,24 +311,24 @@ function updateConfig (changes, config) {
   const configuration = []
 
   for (const change of changes) {
-    if (!names.hasOwnProperty(change.name)) continue
-
-    const name = names[change.name]
+    const name = names[change.name] || change.name
     const { origin, value } = change
-    const entry = { name, origin, value }
+    const entry = { name, value, origin }
 
-    if (Array.isArray(value)) {
-      entry.value = value.join(',')
-    } else if (name === 'DD_TAGS') {
-      entry.value = Object.entries(value).map(([key, value]) => `${key}:${value}`)
-    }
+    if (Array.isArray(value)) entry.value = value.join(',')
+    if (entry.name === 'DD_TAGS') entry.value = formatMapForTelemetry(entry.value)
+    if (entry.name === 'url' && entry.value) entry.value = entry.value.toString()
+    if (entry.name === 'appsec.rules') entry.value = JSON.stringify(entry.value)
```

Comment:
@uurien Isn't this irreleveant ? don't we read the rules directly in the appsec loading ? this value should just be a path no ?

Reply:
Yes, we used to have whole rules in `appsec.rules` but we changed it in November. 
cc @iunanua who made the change. `appsec.rules` should be a string.

Reply:
@ida613 
```suggestion
```